cpp/oneapi/dal/backend/primitives/reduction/functors.hpp

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <cmath>
+#include <cstdint>
+
+namespace oneapi::dal::backend::primitives {
+
+template <typename T>
+struct unary_functor {
+    inline T operator()(T arg);
+};
+
+template <typename T>
+struct identity : public unary_functor<T> {
+    inline T operator()(T arg) const {
```

Comment:
`inline` is not needed, unless you want force inlining of these functions, consider use of `ONEDAL_FORCEINLINE`. The same for other functors.

Reply:
won't fix

Reply:
Fixed